portal/src/graphql/portal/ModifyOAuthClientForm.tsx

```diff
@@ -0,0 +1,216 @@
+import React, { useContext, useEffect, useMemo } from "react";
+import cn from "classnames";
+import { DirectionalHint, TagPicker, Text, TextField } from "@fluentui/react";
+import { Context } from "@oursky/react-messageformat";
+
+import LabelWithTooltip from "../../LabelWithTooltip";
+import { useTagPickerWithNewTags, useTextField } from "../../hook/useInput";
+import { OAuthClientConfig } from "../../types";
+import { parseError } from "../../util/error";
+import { defaultFormatErrorMessageList } from "../../util/validation";
+
+import styles from "./ModifyOAuthClientForm.module.scss";
+
+interface ModifyOAuthClientFormProps {
+  className?: string;
+  clientConfig: OAuthClientConfig;
+  onClientConfigChange: (newClientConfig: OAuthClientConfig) => void;
+  updateAppConfigError: unknown;
+}
+
+function constructClientConfigState(
+  clientId: string,
+  accessTokenLifetime: string,
+  refreshTokenLifetime: string,
+  redirectUris: string[],
+  postLogoutRedirectUris: string[]
+): OAuthClientConfig {
+  // Number("") = 0
+  const accessTokenLifetimeSec = Number(accessTokenLifetime);
+  const refreshTokenLifetimeSec = Number(refreshTokenLifetime);
+
+  return {
+    client_id: clientId,
+    redirect_uris: redirectUris,
+    post_logout_redirect_uris:
+      postLogoutRedirectUris.length > 0 ? postLogoutRedirectUris : undefined,
+    grant_types: ["authorization_code", "refresh_token"],
```

Comment:
Sorry I missed one grant type for anonymous user: `urn:authgear:params:oauth:grant-type:anonymous-request`. Let's add that as well. cc @louischan-oursky